storage_s3/indico_storage_s3/plugin.py

```diff
@@ -0,0 +1,148 @@
+        self.client = boto3.client(
+            's3',
+            endpoint_url=self.host,
+            aws_access_key_id=self.access_key,
+            aws_secret_access_key=self.secret_key,
+        )
+
+    def open(self, file_id):
+        try:
+            return self.client.get_object(Bucket=self._get_current_bucket(), Key=file_id)['Body']
+        except Exception as e:
+            raise StorageError('Could not open "{}": {}'.format(file_id, e)), None, sys.exc_info()[2]
+
+    @contextmanager
+    def get_local_path(self, file_id):
+        with NamedTemporaryFile(suffix='indico.tmp', dir=config.TEMP_DIR) as tmpfile:
+            self._copy_file(self.open(file_id), tmpfile)
+            tmpfile.flush()
+            yield tmpfile.name
+
+    def save(self, name, content_type, filename, fileobj):
+        try:
+            fileobject = self._ensure_fileobj(fileobj)
+            bucket = self._get_current_bucket()
+            self.client.upload_fileobj(fileobject, bucket, name)
+            checksum = self.client.head_object(Bucket=bucket, Key=name)['ETag'][1:-1]
+            return name, checksum
+        except Exception as e:
+            raise StorageError('Could not save "{}": {}'.format(name, e)), None, sys.exc_info()[2]
+
+    def delete(self, file_id):
+        try:
+            self.client.delete_object(self._get_current_bucket(), file_id)
+        except Exception as e:
+            raise StorageError('Could not delete "{}": {}'.format(file_id, e)), None, sys.exc_info()[2]
+
+    def getsize(self, file_id):
+        try:
+            return self.client.head_object(Bucket=self._get_current_bucket(), Key=file_id)['ContentLength']
+        except Exception as e:
+            raise StorageError('Could not get size of "{}": {}'.format(file_id, e)), None, sys.exc_info()[2]
+
+    def send_file(self, file_id, content_type, filename, inline=True):
+        try:
+            content_disp = ('inline' if inline else 'attachment')
+            h = Headers()
+            h.add('Content-Disposition', content_disp, filename=filename)
+            url = self.client.generate_presigned_url('get_object',
+                                                     Params={'Bucket': self._get_current_bucket(),
+                                                             'Key': file_id,
+                                                             'ResponseContentDisposition': h.get('Content-Disposition'),
+                                                             'ResponseContentType': content_type},
+                                                     ExpiresIn=120)
+            return redirect(url)
+        except Exception as e:
+            raise StorageError('Could not send file "{}": {}'.format(file_id, e)), None, sys.exc_info()[2]
+
+    def _get_current_bucket(self):
+        if any(x in self.bucket for x in ['<year>', '<month>', '<week>']):
```

Comment:
IS the check actually needed? `_replace_bucket_placeholders` doesn't do anything if there are no placeholders anyway.